addOns/retest/src/main/java/org/zaproxy/addon/retest/RetestDialog.java

```diff
@@ -0,0 +1,486 @@
+                            extAutomation.registerPlan(plan);
+                            extAutomation.loadAndRunPlanPanel(plan, true, false);
+                            cancelButton.doClick();
+                        }
+                    });
+        }
+        return createButton;
+    }
+
+    private JButton getCancelButton() {
+        if (cancelButton == null) {
+            cancelButton = new JButton();
+            cancelButton.setText(Constant.messages.getString("retest.dialog.button.cancel"));
+            cancelButton.addActionListener(e -> clearAndCloseDialog());
+        }
+        return cancelButton;
+    }
+
+    private void clearAndCloseDialog() {
+        getPlanTableModel().clear();
+        dispose();
+    }
+
+    public void addAlert(Alert alert) {
+        Alert copyAlert = alert.newInstance();
+        getPlanTableModel().addRow(copyAlert);
+    }
+
+    @Override
+    public void eventReceived(Event event) {
+        SwingUtilities.invokeLater(() -> handleEvent(event));
+    }
+
+    private void handleEvent(Event event) {
+        if (event.getEventType().equals(AutomationEventPublisher.JOB_FINISHED)) {
+            ExtensionAutomation extAutomation =
+                    Control.getSingleton()
+                            .getExtensionLoader()
+                            .getExtension(ExtensionAutomation.class);
+            // TODO
+            // Tidy up when the TEST_FINISHED event changes are there
+            AutomationJob job = extAutomation.getJobByEvent(event);
+            ArrayList<AlertData> alertDataList = new ArrayList<>(getPlanTableModel().getAllRows());
+            if (job.getType().equals(ActiveScanJob.JOB_NAME)
+                    || job.getType().equals(PassiveScanWaitJob.JOB_NAME)) {
+                List<AbstractAutomationTest> alertTests = job.getTests();
+                for (AbstractAutomationTest test : alertTests) {
+                    AutomationAlertTest.Data testData = ((AutomationAlertTest) test).getData();
+                    for (int i = 0; i < alertDataList.size(); i++) {
+                        AlertData data = alertDataList.get(i);
+                        if (data.getScanRuleId() == testData.getScanRuleId()
+                                && data.getAlertName().equals(testData.getAlertName())
+                                && data.getUrl().equals(testData.getUrl())
+                                && data.getMethod().equals(testData.getMethod())
+                                && data.getAttack().equals(testData.getAttack())
+                                && data.getParam().equals(testData.getParam())
+                                && data.getEvidence().equals(testData.getEvidence())
+                                && data.getConfidence().equals(testData.getConfidence())
+                                && data.getRisk().equals(testData.getRisk())
+                                && data.getOtherInfo().equals(testData.getOtherInfo())) {
```

Comment:
Is there not an Alert.equals method? Or Alert.getFingerPrint (which could then be compared)?

Reply:
This was discussed during a call, since it's individual elements it doesn't really make sense to instantiate a new Alert for the comparison.